crates/sui-adapter/src/execution_engine.rs

```diff
@@ -382,6 +382,7 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
             vec![
                 system_object_arg,
                 CallArg::Pure(bcs::to_bytes(&change_epoch.epoch).unwrap()),
+                CallArg::Pure(bcs::to_bytes(&change_epoch.protocol_version).unwrap()),
```

Comment:
It's a bit strange to me that we only pass this on safe mode when we fail to advance normally but not the regular mode.

Reply:
As we discussed on slack, I'm ok with doing it in both places. Will fix.

Reply:
i'm fairly alarmed the tests passed without this.

Reply:
wait, really?

Reply:
as far as i can tell, yes - i wonder if we are always hitting safe mode reconfig in the tests when this happens.